packages/concerto-core/lib/introspect/property.js

```diff
@@ -178,15 +179,15 @@ class Property extends Decorated {
 
         const parent = this.getParent();
         if(!parent) {
-            throw new Error('Property ' + this.name + ' does not have a parent.');
+            throw new IllegalModelException('Property ' + this.name + ' does not have a parent.');
```

Comment:
This wouldn't close #834 since these exceptions are not adding any error codes. The whole point of [this](https://github.com/accordproject/concerto/pull/773/files), was to not rely on just the types of exceptions.

Reply:
hey Should I make a new error code for `IllegalModelException` like the sample pr you mentioned or should I use the remaining one from errorCodes file. @ekarademir

Reply:
Hey I have added a new param called `errorType` in illeagalexception and basefilexception , 
Is this the correct way to add error codes? need some feedback @ekarademir

Reply:
They should follow the class name of the exception being thrown.